tests/foreman/cli/test_activationkey.py

```diff
@@ -1452,3 +1455,115 @@ def test_positive_remove_user(self):
+            'Test_*_{0}'.format(gen_string('alpha'))
+        )
+        ak_name = '{0}_{1}'.format(ak_name_like, gen_string('alpha'))
+        org = make_org()
+        self.upload_manifest(org['id'], manifests.clone())
+        available_subscriptions = Subscription.list(
+            {'organization-id': org['id']}, per_page=False)
+        self.assertGreater(len(available_subscriptions), 0)
+        available_subscription_ids = [
+            subscription['id']
+            for subscription in available_subscriptions
+        ]
+        subscription_id = choice(available_subscription_ids)
+        activation_key = self._make_activation_key({
+            'name': ak_name,
+            'organization-id': org['id']
+        })
+        ActivationKey.add_subscription({
+            u'id': activation_key['id'],
+            u'subscription-id': subscription_id,
+        })
+        subscriptions = ActivationKey.subscriptions({
+            'organization-id': org['id'],
+            'id': activation_key['id'],
+        }, output_format='csv')
+        self.assertEqual(len(subscriptions), 1)
+        role = make_role({'organization-id': org['id']})
+        resource_permissions = {
+            'Katello::ActivationKey': {
+                'permissions': [
+                    'view_activation_keys',
+                    'create_activation_keys',
+                    'edit_activation_keys',
+                    'destroy_activation_keys'
+                ],
+                'search': "name ~ {}".format(ak_name_like)
+            },
+            'Katello::HostCollection': {
+                'permissions': [
+                    'view_host_collections',
+                    'edit_host_collections'
+                ],
+                'search': "name ~ {0} || name ~ {1}".format(*hc_names_like)
+            },
+            'Organization': {
+                'permissions': [
+                    'view_organizations',
+                    'assign_organizations',
+                ],
+            },
+            'Katello::Subscription': {
+                'permissions': [
+                    'view_subscriptions',
+                    'attach_subscriptions',
+                    'unattach_subscriptions'
+                ],
+            }
+        }
+        add_role_permissions(role['id'], resource_permissions)
+        user = make_user({
+            'admin': False,
+            'default-organization-id': org['id'],
+            'organization-ids': [org['id']],
+            'login': user_name,
+            'password': user_password,
+        })
+        User.add_role({'id': user['id'], 'role-id': role['id']})
+        ak_user_cli_session = ActivationKey.with_user(user_name, user_password)
+        subscriptions = ak_user_cli_session.subscriptions({
+            'organization-id': org['id'],
+            'id': activation_key['id'],
+        }, output_format='csv')
+        self.assertEqual(len(subscriptions), 1)
```

Comment:
And what if empty string is returned? 🙂 
Please also add a check that actual subscription is present in output (probably by subscription name).

Reply:
format csv, this a table format I think, anyway if empty string is return the assertion will fail, will add to check that the subscription id is the same

Reply:
```python
In [3]: subscriptions = ['']

In [4]: len(subscriptions)
Out[4]: 1
```
What do you mean? It will pass the check. Btw it's pretty common output - for some commands extra empty string is returned, so in case command returned nothing one empty string is present. Not sure about specific commands affected by the issue, but apparently there are some.

Reply:
any way this command ha no empty string the id checking is added that will prevent passing in such case, see updated code